clippy_lints/src/redundant_closure_call.rs

```diff
@@ -136,6 +144,23 @@ fn get_parent_call_exprs<'tcx>(
     (expr, depth)
 }
 
+/// This has been pulled from `in_external_macro`, and the checks for being outside the
+/// current crate have been removed.
+fn in_macro(expr: &'_ hir::Expr<'_>) -> bool {
+    let expn_data = expr.span.ctxt().outer_expn_data();
+    !matches!(
+        expn_data.kind,
+        ExpnKind::Root
+            | ExpnKind::Desugaring(
```

Comment:
Can we just match `ExpnKind::Macro` here?

Reply:
Yes we can!